mypy/plugin.py

```diff
@@ -81,6 +117,12 @@ def get_method_signature_hook(self, fullname: str) -> Optional[MethodSignatureHo
     def get_method_hook(self, fullname: str) -> Optional[MethodHook]:
         return None
 
+    def get_attribute_hook(self, fullname: str) -> Optional[AttributeHook]:
```

Comment:
Redesigned the plugin system to not rely on complex callback types. Now each callback just takes a single argument. The type of the argument depends on the kind of the callback, but I think that it's easy enough to keep track of a single argument type per callback. Callback types are now simple enough to not require type aliases. This also makes it easier to pass more stuff to callbacks in the future, as existing callbacks will likely continue to work, whereas if we'd add new arguments, existing callbacks would break.